packages/cli/src/cmds/init/initHandler.test.ts

```diff
@@ -1,584 +1,296 @@
 import { promises as fs } from 'fs';
-import {
-  getSnapSourceShasum,
-  NpmSnapFileNames,
-  getWritableManifest,
-} from '@metamask/snap-utils';
-import mkdirp from 'mkdirp';
+import pathUtils from 'path';
 import * as snapUtils from '@metamask/snap-utils';
 import {
   getPackageJson,
   getSnapManifest,
 } from '@metamask/snap-utils/test-utils';
-import * as miscUtils from '../../utils/misc';
-import * as readlineUtils from '../../utils/readline';
-import { TemplateType } from '../../builders';
-import template from './init-template.json';
-import { initHandler, updateManifestShasum } from './initHandler';
+import { YargsArgs } from '../../types/yargs';
+import { initHandler } from './initHandler';
 import * as initUtils from './initUtils';
 
-jest.mock('mkdirp');
 jest.mock('@metamask/snap-utils');
 
-const mkdirpMock = mkdirp as unknown as jest.Mock;
-
 const getMockArgv = () => {
   return {
-    dist: 'dist',
-    outfileName: 'bundle.js',
-    src: 'src/index.js',
-    port: 8081,
+    directory: 'foo',
   } as any;
 };
 
 describe('initialize', () => {
   describe('initHandler', () => {
-    beforeEach(() => {
-      jest
-        .spyOn(initUtils, 'buildSnapManifest')
-        .mockImplementation(async () => [getSnapManifest(), getMockArgv()]);
-      jest.spyOn(initUtils, 'prepareWorkingDirectory').mockImplementation();
-      jest.spyOn(console, 'log').mockImplementation();
-      jest.spyOn(readlineUtils, 'closePrompt').mockImplementation();
-    });
-
     afterEach(() => {
       global.snaps = {};
     });
 
-    it('successfully initializes a Snap project', async () => {
-      const fsWriteMock = jest.spyOn(fs, 'writeFile').mockImplementation();
+    it('successfully initializes a Snap project in the current working directory', async () => {
```

Comment:
Rather than mocking everything, could we use `memfs` for testing? See `utils` for example: https://github.com/MetaMask/snaps-monorepo/blob/554f3b40d1dba34972184c5907b9fc4b82de1497/packages/utils/src/manifest.test.ts#L26-L48